src/pages/index.js

```diff
@@ -1,56 +1,71 @@
 import { Card } from "../components/Card.js";
 import { FormValidator } from "../components/FormValidator.js";
 import { Section } from "../components/Section.js";
-
 import { PopupWithForm } from "../components/PopupWithForm.js";
 import { PopupWithImage } from "../components/PopupWithImage.js";
-
+import { PopupWithConfirmation } from "../components/PopupWithConfirmation.js";
 import { UserInfo } from "../components/UserInfo.js";
+import { api } from "../components/Api.js";
 
 import "./index.css";
-
-import {
-  initialCards,
-  selectors,
-  formValidationSettings,
-} from "../utils/constants.js";
+import { selectors, formValidationSettings } from "../utils/constants.js";
 
 const {
   profileModalSelector,
   placeModalSelector,
   imageModalSelector,
   profileNameSelector,
   profileJobSelector,
+  profileImageSelector,
   cardsListSelector,
+  deleteCardModalSelector,
+  editAvatarModalSelector,
 } = selectors;
 
 const editProfileButton = document.querySelector(".profile__edit-button");
 const addPlaceButton = document.querySelector(".profile__add-button");
+const editAvatarElement = document.querySelector(".profile__avatar-container");
+
+const formValidators = {};
 
 const getCardElement = (item) => {
-  const card = new Card(item, "#card-template", handleImageClick);
-  const cardElement = card.generateCard();
-  return cardElement;
+  const card = new Card(
+    item,
+    "#card-template",
+    handleImageClick,
+    openDeleteCardPopup,
+    handleLikeClick
+  );
+  return card.generateCard();
 };
 
 const cardsList = new Section(
   {
-    items: initialCards,
-    renderer: (item) => {
-      const cardElement = getCardElement(item);
-      cardsList.addItem(cardElement, { method: "prepend" });
-    },
+    renderer: (item) => getCardElement(item),
   },
   cardsListSelector
 );
 
-cardsList.renderItems();
+const userInfo = new UserInfo({
+  nameSelector: profileNameSelector,
+  jobSelector: profileJobSelector,
+  profileImageSelector: profileImageSelector,
+});
+
+api.getData().then(({ userInfo: userData, cards }) => {
+  const { name, about, avatar } = userData;
+  userInfo.setUserInfo({
+    name,
+    job: about,
+    avatarUrl: avatar,
+  });
+  cardsList.renderItems(cards.reverse());
+});
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**